src/reducers/computeRenderedRows.js

```diff
@@ -139,7 +140,11 @@ function calculateRenderedRowRange(state, scrollAnchor) {
     rowIdx >= 0 &&
     totalHeight < maxAvailableHeight
   ) {
-    totalHeight += updateRowHeight(state, rowIdx);
+    if (state.isVerticalScrollExact) {
+      totalHeight += rowOffsetIntervalTree.get(rowIdx);
+    } else {
+      totalHeight += updateRowHeight(state, rowIdx);
+    }
```

Comment:
Are we checking for `isVerticalScrollExact` just to avoid fetching the rows twice in the scenario where `initializeRowHeightsAndOffsets()` is already called?
If so, `initializeRowHeightsAndOffsets()` is only called when the no:of total rows or default row height changes.
So the new change effectively can lead to stale row heights for visible rows when `isVerticalScrollExact` is turned on.
This is a change in behavior where the visible row heights were always updated on the fly.

IMO, we don't need this change cause we're only re-fetching visible rows and that's not a big deal in the grand scale of things.
If we really need this micro-optimization, another approach _might_ be to make `updateRowHeight()` smarter, and have it only fetch the row height once per render. But again, this can be done later.

Reply:
yeah i would also suggest to keep the same behaviour and avoid this optimization for now.

Reply:
Indeed, the ```isVerticalScrollExact``` was just added to avoid fetching the row heights twice. 
I have removed this optimization